hydromt/models/components/base.py

```diff
@@ -12,7 +12,7 @@
 
 
 class ModelComponent:
-    """Base Model component class."""
+    """Abstract base class for ModelComponent."""
 
     def __init__(self, model: "Model"):
         self._model_ref: ReferenceType["Model"] = ref(model)
```

Comment:
Before merging, you had added a couple of abstractmethod for set / read /write.
I think maybe we still want that in case users want to implement their own component then it's clear what methods they need to add.
What we can do is open an issue to finalise ModelComponent together with region in a different PR.
@savente93 already created a couple of issues for region that should be solved to get everything working properly.
Maybe @savente93 could you open a more general issue that recaps all the steps that are needed to finalize ModelComponent implementation before we are then ready to add new ones? Then you can mention there the abstractmethod of ModelComponent.

Then @Tjalling-dejong you can leave it as is for this PR.

Reply:
I think this should get solved when Marnix and I work on integrating the components with the model itself since that is when that API really has to become concrete. I'll make a ticket if you like, but I think this is okay as is in this PR